src/raygun.rum/index.js

```diff
@@ -499,8 +505,27 @@ var raygunRumFactory = function (window, $, Raygun) {
     }
 
     function sendCoreWebVitalTimings(performanceData) {
-      // Core web vital timing metrics need to be sent to the API immediately, if they are queued then they may not be sent if a virtual page timing event occurs before they are tracked
```

Comment:
Is the issue of a virtual page occurring resolved with this new approach?

Reply:
Yes, we changed how Core web vitals are sent by creating a new handler specifically for them. This avoids any potential queuing race conditions with other events so CWV will always be sent immediately when they occur

Reply:
Hamish covered this one above - to add some extra context the new behaviour will attempt to send all web vital events using the send beacon API from inside this function. This is to mitigate the browser cancelling events when a request is fired at the same time the browser is attempting to navigate (causing all incomplete events to cancel at the time of navigation). We added additional fallbacks so that if `sendBeacon` is unavailable just use the regular queued approach to provide some level of backwards compatibility support based on the previous functionality